org.mixedrealitytoolkit.uxcore/Experimental/NonNativeKeyboard/NonNativeKeyboardTouchAdapter.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Mixed Reality Toolkit Contributors
+// Licensed under the BSD 3-Clauseusing System;
+
+using Microsoft.MixedReality.GraphicsTools;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace MixedReality.Toolkit.UX.Experimental
+{
+    public class NonNativeKeyboardTouchAdapter : MonoBehaviour
+    {
+        private void Awake()
+        {
+            var defaultAudioComponent = GetComponent<AudioSource>();
+            defaultAudioComponent.playOnAwake = false;
+            defaultAudioComponent.spatialize = false;
+        }
+
+        private void Start()
+        {
+            var buttons = GetComponentsInChildren<Button>(true);
+            foreach (var button in buttons)
+            {
+                // The search box has an incorrect collider and should not act as a button anyway
+                if (button.gameObject.name != "search")
```

Comment:
This seems a bit hacky...querying for "search".

Why can't you explicitly add the "NonNativeKeyTouchAdapter" to the buttons in the keyboard?  That would remove the need for this class.

Reply:
The buttons are not a prefab. In theory I an do what you want. Manually add them, one by one, for every key. I thought the code solution was pretty neat